dv/cs_registers/tb_cs_registers.core

```diff
@@ -0,0 +1,83 @@
+CAPI=2:
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+name: "lowrisc:ibex:tb_cs_registers"
+description: "CS registers testbench"
+filesets:
+  files_sim_verilator:
+    depend:
+      - lowrisc:dv_verilator:simutil_verilator
+    files:
+      - tb/tb_cs_registers.cc: { file_type: cppSource }
+
+  files_sim:
+    depend:
+      - lowrisc:ibex:ibex_core
+    files:
+      - env/env_dpi.sv
+      - rst_driver/rst_dpi.sv
+      - reg_driver/reg_dpi.sv
+      - tb/tb_cs_registers.sv
+    file_type: systemVerilogSource
+
+parameters:
+  PMPEnable:
+    datatype: bool
+    paramtype: vlogparam
+    default: true
+    description: PMP enabled [true/false]
+  PMPNumRegions:
+    datatype: int
+    paramtype: vlogparam
+    default: 4
+    description: Number of implemented PMP regions [0/16]
+  PMPGranularity:
+    datatype: int
+    paramtype: vlogparam
+    default: 0
+    description: Minimum PMP matching granularity [0/31]
+
+targets:
+  sim:
+    default_tool: verilator
+    toplevel: tb_cs_registers
+    filesets:
+      - files_sim
+      - files_sim_verilator
+    parameters:
+      - PMPEnable
+      - PMPNumRegions
+      - PMPGranularity
+
+    tools:
+      vcs:
+        vcs_options:
+          - '${PWD}/build/bin/reg_dpi.so'
+          - '-debug_access+all'
+
+      verilator:
+        mode: cc
+        libs:
+          - '${PWD}/build/bin/reg_dpi.so'
+        verilator_options:
+# Disabling tracing reduces compile times by multiple times, but doesn't have a
+# huge influence on runtime performance. (Based on early observations.)
+          - '--trace'
+          - '--trace-fst' # this requires -DVM_TRACE_FMT_FST in CFLAGS below!
+          - '--trace-structs'
+          - '--trace-params'
+          - '--trace-max-array 1024'
```

Comment:
I'll leave that up to you, but you can keep fusesoc as the entry point to running the simulation, as opposed to the Makefile, by using hooks. (Both approaches work and I don't see a clear winner on the "cleanest" approach.)

```
# register script (top-level YAML, same level as targets)
scripts:
  compile_dpi_modules:
    # all scripts are called form the work root (sim-verilator). 
    cmd : [make -C some/dir]

# hook registration in the target
targets:
  sim:
    tools:
      ...
    hooks:
      pre_build:
        - compile_dpi_modules
```

(You need a fusesoc >= 1.9.3 for that.)